spotify-artist-viewer/Models/SAArtistViewController.m

```diff
@@ -8,46 +8,63 @@
 
 #import "SAArtistViewController.h"
 #import "SAArtist.h"
+#import "SARequestManager.h"
 #import <SDWebImage/UIImageView+WebCache.h>
 
 @interface SAArtistViewController ()
 
 @property (weak, nonatomic) IBOutlet UIImageView *artistImg;
+@property (weak, nonatomic) IBOutlet UIButton *backButton;
+@property (weak, nonatomic) IBOutlet UILabel *header;
+@property (weak, nonatomic) IBOutlet UITextView *bio;
 
 @end
 
 @implementation SAArtistViewController
-@synthesize artistNameVC;
-@synthesize header;
-@synthesize bio;
-@synthesize img;
-@synthesize vcArtist;
+
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    header.text = vcArtist.artistName;
-    
-    NSURL *imageURL = [NSURL URLWithString:vcArtist.imgURL];
+    self.header.text = self.vcArtist.artistName;
+    NSURL *imageURL = [NSURL URLWithString:self.vcArtist.imgURL];
     [self.artistImg sd_setImageWithURL:imageURL];
+    self.artistImg.layer.cornerRadius = self.artistImg.frame.size.height/2;
+    [self.artistImg.layer setMasksToBounds:YES];
+
+    UIButton *backButton = [[UIBarButtonItem alloc] initWithTitle:@"Back" style:UIBarButtonItemStyleBordered target:self action:@selector(doneButtonPressed:)];
+    self.navigationItem.backBarButtonItem = backButton;
+ 
+    SARequestManager *manager = [SARequestManager sharedManager];
+    [manager getBio:self.vcArtist.artistUri success:^(NSArray *bios){
+        if(bios.count != 0){
+            self.vcArtist.bio = [bios objectAtIndex:0];
+            dispatch_async(dispatch_get_main_queue(), ^{
+               self.bio.text = self.vcArtist.bio;
+            });
+        }
+        
+    } failure:^(NSError *error) {
+        NSLog(@"block failure");
+    }];
+    self.bio.text = self.vcArtist.bio;
```

Comment:
When this line is run the self.vcArtist.bio is likely nil because the getBio method has not been completed. As a result this line does not do anything, self.bio.text is set in the success block. So remove this line.